From 579ea84de6f7c12a7c2e770aa14d034b2d0affa5 Mon Sep 17 00:00:00 2001
From: Arnaud Patard <arnaud.patard@rtp-net.org>
Date: Tue, 19 Jul 2011 09:50:47 +0200
Subject: [PATCH 2/2] Buildd: fix configuration parsing

When dealing with configuration without take_from_dists, it ends up
calling Buildd::DistConf::new_hash() with a hash ref but new_hash()
is expecting a hash. Fix that by using same kind of arguments as
the one used in the other new_hash() calls.

Signed-off-by: Arnaud Patard <arnaud.patard@rtp-net.org>
---
 lib/Buildd/Conf.pm |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/lib/Buildd/Conf.pm b/lib/Buildd/Conf.pm
index dcf4bd0..2bd5324 100644
--- a/lib/Buildd/Conf.pm
+++ b/lib/Buildd/Conf.pm
@@ -532,7 +532,8 @@ if (\@take_from_dists) {
 	#Make one entry per distribution, it's easier later on:
 	for my \$dist (\@dist_names) {
 	    \$entry{'DIST_NAME'} = \$dist;
-                    my \$dist_config = Buildd::DistConf::new_hash(\\\%entry);
+                    my \$dist_config = Buildd::DistConf::new_hash(CHECK=>$conf->{'CHECK'},
+						      HASH=>\\\%entry);
                     push \@distributions_info, \$dist_config;
 	}
     }
-- 
1.7.4.1

